sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -312,7 +312,15 @@ case class AdaptiveSparkPlanExec(
   }
 
   override def doExecuteBroadcast[T](): broadcast.Broadcast[T] = {
-    getFinalPhysicalPlan().doExecuteBroadcast()
+    getFinalPhysicalPlan() match {
```

Comment:
Why the getFinalPhysicalPlan() is `WholeStageCodegen  `and not `BroadcastQueryStage `+ `WholeStageCodegen  ` ?

Reply:
The root cause of this bug may be the loss of `BroadcastExchange` node in DPP filter. I commit a initial idea in [PR#32741](https://github.com/apache/spark/pull/32741).